design/20220722-publicbundle.md

```diff
@@ -0,0 +1,326 @@
+
+This checklist contains actions which must be completed before a PR implementing this design can be
+merged.
+
+- [ ] This design doc has been discussed and approved
+- [ ] Test plan has been agreed upon and the tests implemented
+- [ ] Feature gate status has been agreed upon (whether the new functionality will be placed behind a feature gate or not)
+- [ ] Graduation criteria is in place if required (if the new functionality is placed behind a feature gate, how will it graduate between stages)
+- [ ] User-facing documentation has been PR-ed against the release branch in [cert-manager/website]
+
+## Summary
+
+Most programs, programming languages and libraries consume their trust bundles from a standard
+location on-disk. On Linux, the standard bundle is typically provided by the distribution; the same
+applies to containers, in which the base image used is often the provider of TLS trust bundle(s).
+
+Currently, `trust` makes it easy to construct a bundle in a mechanical sense, but doesn't really
+help in choosing / collecting the various parts which actually go into a bundle.
+
+We propose that we should start helping in what is likely the most common case: to add an option
+for a "public" trust bundle as a source.
+
+## Motivation
+
+NB: It's worth taking a look at the sources for Debian's [ca-certificates package](https://salsa.debian.org/debian/ca-certificates/-/tree/master/)
+and [Fedora's equivalent](https://src.fedoraproject.org/rpms/ca-certificates), as background on
+how popular Linux distributions package their default certificate bundles.
+
+The ideal situation for the vast majority of applications is likely to be that they don't have to
+explicitly configure their application's trust bundle.
+
+By way of example, a Linux Golang application would [currently](https://github.com/golang/go/blob/2d655fb15a50036547a6bf8f77608aab9fb31368/src/crypto/x509/root_linux.go#L8-L15)
+look in the following locations by default:
+
+```go
+var certFiles = []string{
+	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
+	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
+	"/etc/pki/tls/cacert.pem",                           // OpenELEC
+	"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+	"/etc/ssl/cert.pem",                                 // Alpine Linux
+}
+```
+
+It's a chore to manually specify a location for a trust store, and it's unfamiliar and error-prone
+for many users. However, if a user instead overwrites any of these files with their own
+`trust`-generated certificate trust bundle they'll currently overwrite their public trust bundle
+provided by their OS / container base image.
+
+Unless the user's application only talks to services which the user also controls, what users are
+likely to really want in most cases is a bundle containing both the publicly trusted certs _and_
```

Comment:
What if I want to talk to two different infra providers services and they both have their own distinct roots? I don't think 'public' should be treated as a special case myself as the reality of PKIs in different environments are more nuanced than a single 'internal' root and then 'public' (cross-company communication is another example)

Reply:
> What if I want to talk to two different infra providers services and they both have their own distinct roots?

Then you put the roots of the two providers into configmaps and add them as sources in your `Bundle` resource?